src/tools/kata-ctl/src/ops/volume_ops.rs

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) 2022 Boston University
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::args::{DirectVolSubcommand, DirectVolumeCommand};
+
+use anyhow::{anyhow, Ok, Result};
+use futures::executor;
+use kata_types::mount::{
+    DirectVolumeMountInfo, KATA_DIRECT_VOLUME_ROOT_PATH, KATA_MOUNT_INFO_FILE_NAME,
+};
+use nix;
+use reqwest::StatusCode;
+use safe_path;
+use std::{fs, path::PathBuf, time::Duration};
+use url;
+
+use agent::ResizeVolumeRequest;
+use runtimes;
+
+const TIMEOUT: Duration = Duration::from_millis(200);
+const CONTENT_TYPE_JSON: &str = "application/json";
+
+pub fn handle_direct_volume(vol_cmd: DirectVolumeCommand) -> Result<()> {
+    if !nix::unistd::Uid::effective().is_root() {
+        return Err(anyhow!(
+            "super-user privileges are required for the direct-volume subcommand"
+        ));
+    }
+    let command = vol_cmd.directvol_cmd;
+    let cmd_result: Option<String> = match command {
```

Comment:
Although these functions are not publicly visible, you should consider another function name like `direct_vol_add()` since this is not a member method and `add()` is too general.